nix/nixos/cardano-node.nix

```diff
@@ -36,12 +37,32 @@ in {
         '';
       };
 
-      system-start-time = mkOption {
-        type = types.nullOr types.str;
+      genesis-file = mkOption {
+        type = types.path;
+        description = ''
+          Genesis json file.
+        '';
+      };
+
+      genesis-hash = mkOption {
+        type = types.str;
+        description = ''
+          Hash of the genesis file.
+        '';
+      };
+
+      signing-key = mkOption {
+        type = types.path;
+        description = ''
+          Signing key
+        '';
+      };
+
+      delegation-certificate = mkOption {
```

Comment:
should the delegation certificate be mandatory?

Reply:
It's only required for core nodes that sign blocks. We want to run relays too.